File: NETMouse - .NET release/Examples/PascalABC.NET/Matrices_3.pas
{$reference ABCNET.dll}
uses ABCNET.Utils;
uses ABCNET.Extensions;
uses System;

function Min(self: array [,] of integer): integer; extensionmethod;
begin
  if self = nil then
    raise new ArgumentNullException('self');

  result := integer.MaxValue;
  var rowsCount := self.GetLength(0);
  var colsCount := self.GetLength(1);
  for var i := 0 to rowsCount - 1 do
    for var j := 0 to colsCount - 1 do
      if result > self[i, j] then
        result := self[i, j];
end;

begin
  Matr.Rand(Base.ReadInteger('N:'), Base.ReadInteger('M:'), -30, 45).Print().Min().Println();
end.

File: NETMouse - .NET release/Examples/C#/Matrices_3.cs
using ABCNET.Extensions;
using ABCNET.Utils;
using System;

namespace TestProgram
{
    internal static class Program
    {
        public static int Min(this int[,] matrix)
        {
            if (matrix == null)
                throw new ArgumentNullException("matrix");

            int min = int.MaxValue;
            int rowsCount = matrix.GetLength(0);
            int colsCount = matrix.GetLength(1);
            for (int i = 0; i < rowsCount; i++)
                for (int j = 0; j < colsCount; j++)
                    if (min > matrix[i, j])
                        min = matrix[i, j];

            return min;
        }

        private static void Main(string[] args)
        {
            Matr.Rand(Base.ReadInteger("N:"), Base.ReadInteger("M:"), -30, 45).Print().Min().Println();
        }
    }
}